dbmigrator/tests/test_utils.py

```diff
@@ -36,6 +36,38 @@ def tearDown(self):
                 cursor.execute('DROP TABLE IF EXISTS a_table')
                 cursor.execute('DROP TABLE IF EXISTS schema_migrations')
 
+    def test_super_user(self):
+        from ..utils import super_user, with_cursor
+
+        def cleanup():
+            with psycopg2.connect(testing.db_connection_string,
+                                  user='postgres') as db_conn:
+                with db_conn.cursor() as cursor:
+                    cursor.execute('DROP TABLE IF EXISTS super_table')
+
+        self.addCleanup(cleanup)
+
+        with psycopg2.connect(testing.db_connection_string) as db_conn:
+            with db_conn.cursor() as cursor:
+                cursor.execute('SELECT current_user')
+                self.assertEqual('travis', cursor.fetchone()[0])
+
+                with super_user() as super_cursor:
+                    super_cursor.execute('SELECT current_user')
+                    self.assertEqual('postgres', super_cursor.fetchone()[0])
+
+                    super_cursor.execute('CREATE TABLE super_table (id TEXT)')
+                    super_cursor.execute(
+                        'GRANT ALL PRIVILEGES ON TABLE super_table TO travis')
+                    super_cursor.execute(
+                        "INSERT INTO super_table VALUES ('62')")
+
+                cursor.execute('SELECT current_user')
+                self.assertEqual('travis', cursor.fetchone()[0])
+
+                cursor.execute('SELECT * FROM super_table')
```

Comment:
@reedstrm I added this to make sure the table created in the context manager can be accessed outside of it.  It appears to be working.  Is there a specific case when it doesn't work?

Reply:
Hmm, I'll look into this. Was pretty sure I had run into it.